task_processing/plugins/kubernetes/kubernetes_pod_executor.py

```diff
@@ -394,7 +432,24 @@ def reconcile(self, task_config: KubernetesTaskConfig) -> None:
         # TASKPROC(244): we probably only want reconcile() to fill in the task_config
         # member of the relevant KubernetesTaskMetadata and do an unconditional state
         # reconciliation on startup
-        pass
+        pod_name = task_config.pod_name
+        pod = self.kube_client.get_pod(namespace=self.namespace, pod_name=pod_name)
+
+        task_metadata = self.task_metadata[pod_name]
+        self.task_metadata.set(
+            pod_name,
+            task_metadata.set(
+                task_config=task_config
+            )
+        )
+
+        if not pod:
+            # Pod has gone away while restarting
+            logger.info(f"Pod {pod_name} for task {task_config.name} was no longer found.")
+            self.__update_deleted_pod(pod_name=pod_name, event=None)
```

Comment:
I think we'll want to add a helper for this case and send a `lost` event instead of treating this like a deleted pod event